### PR TITLE
Scale rosa cluster

### PR DESCRIPTION
**Pull request by svetsa-rh**
Trying to create cluster with needed replicas.

### CODE REVIEW

**Comment by svetsa-rh** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-node-density-5nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-cluster-density-v2-5nodes

**Comment by svetsa-rh** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-node-density-5nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-cluster-density-v2-5nodes

**Comment by svetsa-rh** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-node-density-5nodes

**Comment by svetsa-rh** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-node-density-5nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-cluster-density-v2-5nodes

**Comment by svetsa-rh** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-node-density-5nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-cluster-density-v2-5nodes

**Comment by svetsa-rh** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-24workernodes

**Comment by svetsa-rh** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-cluster-density-24workernodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-cluster-density-v2-24nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-node-density-24nodes

**Comment by svetsa-rh** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-cluster-density-24workernodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-cluster-density-v2-24nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-node-density-24nodes

**Comment by svetsa-rh** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-node-density-24nodes

**Comment by svetsa-rh** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.12-nightly-ocp-qe-perfscale-ci-tests-rosa-node-density periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-node-density-24nodes periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.13-nightly-ocp-qe-perfscale-ci-tests-rosa-cluster-density-24nodes

**Comment by svetsa-rh** [comment]
@paigerube14 @qiliRedHat Can you pls check my latest commit in this PR https://github.com/openshift/release/pull/40209/commits/3580ac1c0c22a2f46482d6cd181629656a50d14d and see if it warrants running ocp-qe-perfscale-ci-tests on non rosa environments, such as gcp? 
Or having passing node density tests for rosa env is good enough to verify the changes?

**Comment by paigerube14** [comment]
@svetsa-rh I would think the type of check you added here should be added to the Rosa provisioning workload and shouldn't be in our tests set up specifically. We would also want to check that the number of worker nodes matches the number of replicas because I have seen where machines fail and no node gets created so in your case I think your check would pass even without matching replicas

**Comment by svetsa-rh** [comment]
> @svetsa-rh I would think the type of check you added here should be added to the Rosa provisioning workload and shouldn't be in our tests set up specifically. We would also want to check that the number of worker nodes matches the number of replicas because I have seen where machines fail and no node gets created so in your case I think your check would pass even without matching replicas

Thanks @paigerube14! Agree with both points you mentioned here. Will update accordingly.

**Comment by paigerube14** [comment]
@svetsa-rh maybe ping one of the original creators of the rosa provisioning workflow before adding all this? I'm not sure about the whole negative test path would work in prow. 

In my opinion waiting for all nodes and cluster operators to come up should be additional checks at the cluster creation level. Flexy goes through and waits for all operators to be ready/not degraded. I would think this should be the same. You can also take a look at the aws, azure, or gcp cluster creation scripts and see what all they have as checks there.

**Comment by svetsa-rh** [comment]
@paigerube14 All workers scaling and health checks seem to be done at test suite level and not while provisioning cluster. 

I was already thinking of moving this out of node-density test and have it in a common location so all workloads or other tests can invoke it.

**Comment by paigerube14** [comment]
@svetsa-rh I agree it should be in a place where all workloads could use it, very interesting that its in test suite level currently. Think we should add to all of our "pre" steps

**Comment by qiliRedHat** [comment]
@svetsa-rh I agree with all your discussions above. 
I recently found in my test for classic rosa, after provisioning there were 3 infra nodes created which have 'worker' label as well. This is not the same as our 'normal' cluster. Rosa hypershift doesn't have the 3 infra nodes in provision step.
Those infra nodes will be able to schedule pods because they have 'worker' label, if we don't treat them, actually we will have REPLICAS+3 nodes for our test workloads. I think you will take care of them in post install steps, right?
```
% oc get nodes
NAME                                         STATUS   ROLES                  AGE   VERSION
ip-10-0-132-241.us-west-2.compute.internal   Ready    infra,worker           44m   v1.25.10+3fe2906
ip-10-0-152-169.us-west-2.compute.internal   Ready    worker                 62m   v1.25.10+3fe2906
ip-10-0-152-91.us-west-2.compute.internal    Ready    control-plane,master   73m   v1.25.10+3fe2906
ip-10-0-166-71.us-west-2.compute.internal    Ready    control-plane,master   73m   v1.25.10+3fe2906
ip-10-0-169-231.us-west-2.compute.internal   Ready    infra,worker           44m   v1.25.10+3fe2906
ip-10-0-181-143.us-west-2.compute.internal   Ready    worker                 62m   v1.25.10+3fe2906
ip-10-0-198-192.us-west-2.compute.internal   Ready    control-plane,master   73m   v1.25.10+3fe2906
ip-10-0-203-29.us-west-2.compute.internal    Ready    infra,worker           45m   v1.25.10+3fe2906
ip-10-0-215-180.us-west-2.compute.internal   Ready    worker                 62m   v1.25.10+3fe2906
```

**Comment by paigerube14** [comment]
Looking at some of the past runs as well I am seeing what Qiujie mentioned as well. We'll probably be good to just label them like [this line](https://github.com/openshift-qe/ocp-qe-perfscale-ci/blob/18fe3d3192eaa8e49bd9868031fa48e8e908b9c1/install_infra_work.sh#L193). Also would be good if we could validate if any of the infrastructure (openshift-* namespace pods) are actually already running on the infra nodes by default

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.12-nightly-ocp-qe-perfscale-ci-tests-rosa-network-perf/1669676488732971008/artifacts/ocp-qe-perfscale-ci-tests-rosa-network-perf/openshift-qe-network-smoke/build-log.txt
```
+ log '###############################################'
++ date -u
+ echo -e '\033[1mFri Jun 16 13:20:28 UTC 2023 ###############################################\033[0m'
[1mFri Jun 16 13:20:28 UTC 2023 ###############################################[0m
+ timeout 7200 ./k8s-netperf --debug --metrics --all --config smoke.yaml --search https://search-perfscale-dev-chmf5l4sh66lvxbnadi4bznl3a.us-west-2.es.amazonaws.com:443 --tcp-tolerance 70 --clean=true
time="2023-06-16 13:20:28" level=info msg="📒 Reading smoke.yaml file. "
time="2023-06-16 13:20:28" level=info msg="Cleaning resources created by k8s-netperf"
time="2023-06-16 13:20:28" level=info msg="🔬 prometheus discovered at openshift-monitoring"
time="2023-06-16 13:20:28" level=info msg="Deploying in us-east-1a zone"
time="2023-06-16 13:20:28" level=debug msg="Number of nodes with role worker: 6"
```

**Comment by qiliRedHat** [comment]
@paigerube14 Sorry I have to update about my last comment after our meeting with Sai. 
@svetsa-rh Correct me if 'm wrong with the following statements:
In our meeting with Sai, he said on classic rosa,
1.  they leave the infra nodes with 'worker' label.
2. They don't use workload machinesets, just use one worker node as workload node
3. They don't have to move monitoring and ingress to infra, they are already moved there( I haven't verified this and known where this is done)

**Comment by svetsa-rh** [comment]
@paigerube14 and @qiliRedHat My apologies! I started this PR initially to get scaling of clusters in here, but after Paige's comment https://github.com/openshift/release/pull/40209#issuecomment-1591655278 I started working on it here 
https://github.com/openshift/release/pull/40392

I was waiting for https://github.com/openshift/release/pull/40392 to be merged before proceeding on this. 

Yes, in https://github.com/openshift/release/pull/40392/files#diff-6d4dd4a1c9b08c2b35b45b84097313d8f0e47fe00c3e04650a8f1a7f2ca8c8b4R21 we only account for worker nodes without infra label and compare that number against REPLICAS specified to make sure count matches.

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.13-nightly.yaml [hunk 1]
```diff
@@ -52,6 +52,35 @@ tests:
...
+    cluster_profile: aws-qe
+    env:
+      CHANNEL_GROUP: nightly
+      COMPUTE_MACHINE_TYPE: m5.2xlarge
+      ENABLE_AUTOSCALING: "false"
+      MULTI_AZ: "true"
+      OPENSHIFT_VERSION: "4.13"
+      REPLICAS: "24"
+    test:
+    - ref: openshift-qe-cluster-density
```
Comment (paigerube14) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__rosa-4.13-nightly.yaml:67]: let's change this to cluster-density-v2

**Comment by svetsa-rh** [comment]
Covered as part of https://github.com/openshift/release/pull/40392 and https://github.com/openshift/release/pull/40540
Closing PR.

**Pull request closed without merging**